met-web/src/components/engagement/form/ActionContext.tsx

```diff
@@ -83,59 +74,68 @@ export const ActionProvider = ({ children }: { children: JSX.Element }) => {
     const [savedBannerImageFileName, setSavedBannerImageFileName] = useState('');
     const [contentTabs, setContentTabs] = useState<EngagementContent[]>([createDefaultEngagementContent()]);
     const isCreate = window.location.pathname.includes(CREATE);
+    const noRouteLoaders = { engagement: 'engagement', content: 'content', metadata: 'metadata', taxa: 'taxa' };
+    const { engagement } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { engagement: Promise<Engagement> })
+        : noRouteLoaders;
+    const { content } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { content: Promise<EngagementContent[]> })
+        : noRouteLoaders;
+    const { metadata } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { metadata: Promise<EngagementMetadata[]> })
+        : noRouteLoaders;
+    const { taxa } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { taxa: Promise<MetadataTaxon[]> })
+        : noRouteLoaders;
 
-    const handleAddBannerImage = (files: File[]) => {
-        if (files.length > 0) {
-            setBannerImage(files[0]);
-            return;
-        }
-
-        setBannerImage(null);
-        setSavedBannerImageFileName('');
-    };
-
-    const fetchEngagement = async () => {
+    // Load the engagement from the shared individual engagement loader and watch the engagement variable for any changes.
+    useEffect(() => {
         if (!isCreate && isNaN(Number(engagementId))) {
             navigate('/');
         }
-
-        if (isCreate) {
+        if (isCreate && !engagementId) {
```

Comment:
There is a weird third state for the engagement form (besides create and edit) where the creation form is filled out and the user presses "Save and Continue". The create page is essentially functioning as an edit page. I mentioned this in the new ticket, and that users should just be directly redirected to the edit page when hitting "save and continue".

Reply:
Oh okay I think I see where the weirdness is - because we need to make an early save? And to the user it still appears as if they're "creating" the engagement for the first time. 

It sounds like this is work for the new ticket you were mentioning but in my mind, if the engagement is created, and the form the user uses is functionally identical to the edit form, I feel like we should set `isCreate` to false. Or maybe rename it to something that better illustrates the particular way we create new engagements in the app.